CMakeLists.txt

```diff
@@ -111,7 +111,8 @@ if(MSVC)
   # bigobj: generate more object sections than allowed by default
   # Promoted to errors
   # C4316: object allocated on the heap may not be aligned 16
-  add_compile_options("/wd4244" "/wd4267" "/wd4503" "/wd4800" "/wd4455" "/wd5030" "/bigobj" "/we4316")
+  # /Zc:__cplusplus: correctly set __cplusplus macro value
+  add_compile_options("/wd4244" "/wd4267" "/wd4503" "/wd4800" "/wd4455" "/wd5030" "/bigobj" "/we4316" "/Zc:__cplusplus")
```

Comment:
please note that this option is supported by VC++ 2019 only

Reply:
added a check if(MSVC_VERSION GREATER_EQUAL 1914)